plugin/web/extensions/infobox/core/components/editor/index.tsx

```diff
@@ -0,0 +1,168 @@
+import type { Primitive, PublicSetting, PublicProperty } from "@web/extensions/infobox/types";
+import { Collapse, Button } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import update from "immutability-helper";
+import { useCallback, useEffect, useState } from "react";
+import { DndProvider } from "react-dnd";
+import { HTML5Backend } from "react-dnd-html5-backend";
+
+import type { PropertyItem as PropertyItemType } from "./PropertyItem";
+import PropertyItem from "./PropertyItem";
+
+type Props = {
+  publicSetting: PublicSetting;
+  primitives: Primitive[];
+  savePublicSetting: (publicSetting: PublicSetting) => void;
+};
+
+const Editor: React.FC<Props> = ({ primitives, publicSetting, savePublicSetting, ...props }) => {
+  const [propertyList, setPropertyList] = useState<PropertyItemType[]>([]);
+
+  useEffect(() => {
+    const proptyItems: PropertyItemType[] = [];
+    const primitive = primitives.findLast(p => p.type === publicSetting.type);
```

Comment:
Any reason you are finding the last primitive that meets the requirement? I thought you and I discussed on Wednesday that it made sense to show the values for the currently selected model, no?

Reply:
Currently i assume that new layer system supports multiple select (both of layers and primitives). So consider with multiple primitive selected, here should show the last selected one's value. (And i assume the last selected will be the last in array).

Reply:
Ahhh okay, gotcha